filebeat/module/apache2/access/ingest/default.json

```diff
@@ -4,8 +4,8 @@
     "grok": {
       "field": "message",
       "patterns":[
-        "%{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"%{WORD:apache2.access.method} %{DATA:apache2.access.url} HTTP/%{NUMBER:apache2.access.http_version}\" %{NUMBER:apache2.access.response_code} (?:%{NUMBER:apache2.access.body_sent.bytes}|-)( \"%{DATA:apache2.access.referrer}\")?( \"%{DATA:apache2.access.agent}\")?",
-        "%{IPORHOST:apache2.access.remote_ip} - %{DATA:apache2.access.user_name} \\[%{HTTPDATE:apache2.access.time}\\] \"-\" %{NUMBER:apache2.access.response_code} -"
+        "%{IPORHOST:source.hostname} - %{DATA:user.name} \\[%{HTTPDATE:apache2.access.time}\\] \"%{WORD:http.request.method} %{DATA:url.original} HTTP/%{NUMBER:http.version}\" %{NUMBER:http.response.status_code} (?:%{NUMBER:http.response.body_sent.bytes}|-)( \"%{DATA:http.request.referrer}\")?( \"%{DATA:apache2.access.agent}\")?",
```

Comment:
@webmat The first `IPORHOST` is interesting. Looking at the grok pattern it can be an IP or HOSTNAME (the grok definition of HOSTNAME, no port): https://github.com/elastic/elasticsearch/blob/6.4/libs/grok/src/main/resources/patterns/grok-patterns#L31  So far our test logs seem to contain always IP address here. Could this be different?

If it is a hostname, we would also have an issue further down with running geoip on it I assume. This means probably the pattern here should be modified to just be IP?

Reply:
> %h: Remote hostname. Will log the IP address if HostnameLookups is set to Off, which is the default. If it logs the hostname for only a few hosts, you probably have access control directives mentioning them by name. See the Require host documentation.

You're right that this can be a hostname. Although I would err on the side of not modifying how the module previously worked, otherwise this is no longer a conversion of a module, it becomes an overhaul of the module. And we're talking about doing them all in the next few days...

So in this case, the module probably has problems when people have hostnames there. I would argue this ECS conversion should is unrelated to fixing this problem.

I have no problem if you open issues for problems with this module, though :-)

Reply:
Ok, I just got into a rabbit hole based on this.

So yes, source can be a host (prior to determining if you have an IP literal or a hostname) there. I also saw that `apache2.access.remote_ip` was previously stored as `keyword`, which would have supported this. So unless we decide to change this field name (https://github.com/elastic/ecs/issues/166), `source.hostname` is indeed where you should store this.

Now if this module was doing GeoIP on this field, I guess I would go the extra step and conditionally duplicate this IP to `source.ip` (after confirming this is an IP), and doing GeoIP on that field instead. WDYT?

Reply:
`apache2.access.agent`: why not store directly in `user_agent.original`?

Reply:
Good point about `apache2.access.agent`, changed.